### PR TITLE
Call can_start() and on_pre_start() in the troubleshooting flow

### PR DESCRIPTION
**Pull request by rchl**
Follow to real flow more closely for better error reporting.

### CODE REVIEW

**Comment by rchl** [comment]
Am I blind? What the hell is mypy complaining about (and why I don't see that through `LSP-pylsp`...?

File: plugin/tooling.py [hunk 1]
```diff
@@ -290,10 +290,11 @@ class LspTroubleshootServerCommand(sublime_plugin.WindowCommand, TransportCallba
 
     def run(self) -> None:
         window = self.window
-        active_view = window.active_view()
-        if not active_view:
+        view = window.active_view()
+        if not view:
             sublime.message_dialog('Troubleshooting must be run with a file opened')
             return
+        active_view = view
```
Comment (predragnikolic) [inline plugin/tooling.py:297]: Hahahaha, this was the issue!? 😂

Comment (rchl) [reply plugin/tooling.py:297]: mypy is too stupid to figure out that the variable that we just None-checked is definitively not gonna be `None` so we need to assign it to a new variable.

Comment (rwols) [reply plugin/tooling.py:297]: Actually it _may_ change back to None in the closure, right? The variables are evaluated once the closure is invoked, which is at a later point in time. I'm sure that if you use `functools.partial`, then mypy won't complain. I'm unsure whether mypy is truly correct here. But we did see something similar in the TransportCallbacks in plugin/core/transports.py

Comment (rchl) [reply plugin/tooling.py:297]: I get tripped by similar issue once in a while but I'm not sure that it's the case here.

I don't think that the value of the variable can change in that specific case when lambda before lambda is evaluated. Pyright is smart enough to realize that but not mypy.

If there would be a code after the lambda is created that changes it then I would understand.

**Review by rwols — APPROVED** [review]

**Comment by rwols** [comment]
Please resolve the merge conflicts :)

**Pull request merged**